Lib/test/test_asyncgen.py

```diff
@@ -1169,6 +1169,25 @@ async def run():
 
         self.loop.run_until_complete(run())
 
+    def test_sync_anext_raises_exception(self):
+        # See: https://github.com/python/cpython/issues/131670
+        msg = 'custom'
+        for exc in [
+            StopAsyncIteration(msg),
+            StopIteration(msg),
+            ValueError(msg),
+            Exception(msg),
+        ]:
+            with self.subTest(exc=exc):
+                class A:
+                    def __anext__(self):
+                        raise exc
+
+                with self.assertRaisesRegex(type(exc), msg):
+                    anext(A())
+                with self.assertRaisesRegex(type(exc), msg):
+                    anext(A(), 1)
```

Comment:
```suggestion
        msg = 'custom'
        for exc_type in [
            StopAsyncIteration,
            StopIteration,
            ValueError,
            Exception,
        ]:
            exc = exc_type(msg)
            with self.subTest(exc=exc):
                class A:
                    def __anext__(self):
                        raise exc

                with self.assertRaisesRegex(exc_type, msg):
                    anext(A())
                with self.assertRaisesRegex(exc_type, msg):
                    anext(A(), 1)
```

Maybe only parametrize with the exception type?